waagent

```diff
@@ -2087,6 +2098,24 @@ class FreeBSDDistro(AbstractDistro):
         SetFileContents(location+"/ovf-env.xml", ovfxml)
         return retcode,out
 
+    def fallocate(self,filename,bytes):
+        """
+        Create a non-sparse file of that size. Overwrite existing file. BSD-compatible dd version.
+        
+        To make this a bit faster, a block size of 16M is used. After that, we use a single block to deal with the rest.
+        It might be worth discussing if this should be put here as a method -- probably some `if IsLinux():` is enough.
+        """
+        blocksize = 16 * 1024 * 1024
+        if type(bytes) is str:
+           bytes = int(bytes)
+
+        # We want to make the return value look like real.
+        exitcode = Run("dd if=/dev/zero of=" + filename + " bs=" + str(blocksize) + " count=" + str(bytes/blocksize))
+        if bytes%blocksize:
+            # Finish up using a single block.
+            exitcode = Run("dd if=/dev/zero of=" + filename + " bs=" + str(bytes%blocksize) + " count=1")
```

Comment:
Unless you use "append" this will simply overwrite the first 16MB of the file with more zeroes and not achieve your goal. It would be better to actually pre-compute the correct number of blocks (rounding up if necessary) and just call dd once.